### ISSUE TITLE
Cockpit flight controls not reacting to trim settings.

### ISSUE THREAD

**Issue by Murmur79** [c172p-team/c172p#1471]
If trim settings are modified, the external model correctly shows the control surfaces changing their positions with centered flight controls.

Cockpit flight controls, though, will not react to trim changes, and the yoke and pedals will stay centered, even if the control surfaces are trimmed out-of-center. This is wrong for the C172, since it has direct mechanical link (cables) between flight controls and control surfaces.

The cause of this, is that the three variables:

```
/sim/model/c172p/cockpit/yoke-elevator
/sim/model/c172p/cockpit/yoke-aileron
/sim/model/c172p/cockpit/pedals-rudder
```

are output from:

```
<channel name="Pitch">
<channel name="Roll">
<channel name="Yaw">
```

whereas they should be output from:

```
<summer name="Pitch Trim Sum">
<summer name="Roll Trim Sum">
<summer name="Yaw Trim Sum">
```
which take into account the trim input.

I tried the modification and it works as it should.

**Comment by wlbragg** [comment]
Thank you, we'll take a look and get it merged.

**Comment by wlbragg** [comment]
@dany93 I think this is OK but I would like your approval. If your good with this please go ahead and merge.

**Comment by hbeni** [comment]
Just for the discussion - is this really correct?
If i remember correctly, the trim wheel connects not to the elevator, but it's trim tab.
So trimming when parked for example should have no effect on the elevator - its the airflow flowing over the trim tab that moves the elevator into another position, and this should be eflected/impemented by force-feedback alone.

So in my opinion, the current behaviour is correct regarding the yoke, but the elevator should maybe not be moved by the trim wheel like its currently is (only in flight).

**Comment by Murmur79** [comment]
> Just for the discussion - is this really correct? ~If i remember correctly~, the trim wheel connects not to the elevator, but it's trim tab. So trimming when parked for example should have no effect on the elevator - its the airflow flowing over the trim tab that moves the elevator into another position, and this should be eflected/impemented by force-feedback alone.
> 
> So in my opinion, the current behaviour is correct regarding the yoke, but the elevator should maybe not be moved by the trim wheel like its currently is (only in flight).

You are technically correct, but the current behaviour of trim in most flight simulators is dictated by these 2 issues:
1) PC flight controls are in 99,9% not force-feedback (and hence cannot react to airflow)
2) Even if they were, modeling control surfaces reacting to airflow would be much more difficult to implement anyway - I don't know if the aerodynamic coefficients commonly available from flight tests (Roskam, etc.) include the ones necessary to model airflow effects on control surfaces.

For these reasons, the approach commonly taken by flight simulators is to allow the trim to change flight control displacements even in the absence of airflow. This is IMO the best possible compromise, even if it is technically unrealistic (mostly from a visual point of view, though).

In any case, the current behaviour of FG C172 has certainly a discrepancy, in that external control surfaces move but flight controls remain centered. To keep consistency with how trim action is currently modeled, flight controls should move as well when changing trim settings.

**Comment by dany93** [comment]
I think @hbeni is right regarding reality for the elevator, however @Murmur79's is the best compromise in simulation. The discrepancy is for the elevator yoke position only in flight.

**Comment by hbeni** [comment]
> This is IMO the best possible compromise, even if it is technically unrealistic (mostly from a visual point of view, though).

Thats not just a PoV issue. In real life, you can get out of elevator traversal range when mistrimmed at landings for example. By moving the Yoke also when trimming, you effectively either exaggerate that effect or you inhibit it.

It is a horse lead by a cart: The yoke moves not because of the trim, but because of changed airflow the pressure to the yoke changes. Your model moves the yoke as direct result of trimming.

> In any case, the current behaviour of FG C172 has certainly a discrepancy

Yes it is. However in my opinion the proper fix would be:
1. Add a trim-tab to be animated to the elevator
2. make the elevator trim just displace this trim tab
3. make the yoke control only the elevator _animation_ (not the underlying simulated value for the fdm; that leave as is!)


Maybe I just don't understand correctly...

In any case; I think it is a good Idea to also add a configuration option (probably "on" as default") so the user can disable this in case he has either other feelings or a force feedback system.

I for example fly with mouse, and when trimming the plane the yoke does not need to move. For example when applying power, this results in the correct behavior of "i need to push the yoke forward" and when trimming away the pressure, the yoke can stay where it is. The "felt power to push" goes from _n_ to _0_, but of course i cannot feel this. But it's what happening to the yoke when looking at it when flying.

With a joystick that wants to center itself all the time, this is different, however, because the "no pressure" position is always the centered one, and when letting go of your joystick this will dramatically move your yoke.
Is this fixed by this request here, when properly trimmed?

**Comment by dany93** [comment]
In FG, every FDM that I've seen is such that if the trim is applied on a control, it limits the remaining amplitude for this control surface. If the elevator trim is full, you cannot go farther with the yoke. Even if your joystick can still be moved.

The C172 has no aileron trim, am I right?

**Comment by Murmur79** [comment]
> In FG, every FDM that I've seen is such that if the trim is applied on a control, it limits the remaining amplitude for this control surface. If the elevator trim is full, you cannot go farther with the yoke. Even if your joystick can still be moved.

Indeed, for example in the FG C172, if you trim 100% nose up, pulling the stick will have no effect since the elevator is already at its max nose up position, while pushing the stick all the way down will produce neutral elevator (so you lose 50% of control authority). I think this is the best compromise for a flight sim, given the lack of force feedback on 99.9% of PC flight controls.

The issue however is that currently the trim moves the control surfaces but not the yoke/pedals. This is inconsistent with how trim effects are currently implemented: if it is decided that moving the trim actually moves the elevator (as I said, the best compromise for me), then the yoke should move as well. There is a direct and fixed relationship between flight controls position and flight surfaces position in an aircraft with reversible flight controls like the C172.

As it is now instead, if you move the trim, you can see the elevator/rudder surfaces moving, while the yoke/pedals stay centered. This also happens in flight.

**Comment by hbeni** [comment]
Back in '21 we also had an interesting discussion about this: https://github.com/HHS81/c182s/issues/382

**Comment by Murmur79** [comment]
> Thats not just a PoV issue. In real life, you can get out of elevator traversal range when mistrimmed at landings for example. By moving the Yoke also when trimming, you effectively either exaggerate that effect or you inhibit it.

No, because considering how the trim mechanism is currently implemented in FG (and in most PC flight sims), the modification I'm proposing would not change anything in flight dynamics or control authority.

It would be just a visual change regarding the yoke: as it is now, the yoke always maintains the same centered position when the aircraft is flying trimmed, whatever the speed/AoA it is trimmed at; with my modification instead, the yoke would actually change its position when the aircraft is trimmed for different speeds/AoA's. Just like it happens in the real aircraft.

So, just to be sure, the modification I'm proposing has no effects whatsoever on any aspect of the flight dynamics. It is purely visual, and consistent with how trimming is currently modeled in the sim.

**Comment by hbeni** [comment]
> It would be just a visual change regarding the yoke: as it is now, the yoke always maintains the same centered position when the aircraft is flying trimmed, whatever the speed/AoA it is trimmed for

Really?
This all was stable flight:
(ignore the artificial horizon, i was level. Gyro was mad at me because of CTRL+u(

![grafik](https://github.com/c172p-team/c172p/assets/13608602/64dd3d7f-dac4-4ae6-8da0-5999ef22d622)

![grafik](https://github.com/c172p-team/c172p/assets/13608602/52290ee6-05c6-4053-9f25-0b1a0971791e)

![grafik](https://github.com/c172p-team/c172p/assets/13608602/d5dbc756-eac5-4d2b-89ec-22f720bc9071)

**Comment by Murmur79** [comment]
Again, to reiterate, with my modification nothing would change in flight dynamics, neither for people using joysticks or mouse or any other device.

Indeed, my modification just changes the output to some _visual_ variables (specifically, the position of cockpit flight controls), and it does not change in any way the fdm. This can be easily seen by checking the few changes made to the code.

**Comment by hbeni** [comment]
Thanks for pointing that out :)
And sorry for me being picky :bowtie: :see_no_evil:

**Comment by wlbragg** [comment]
>The C172 has no aileron trim, am I right?

Yes, you are correct. But we do have the elevator trim modeled. I also remember someone working on the FDM to control that surface and vice versa. I don't remember offhand what that entailed though.

**Comment by wlbragg** [comment]
@Murmur79 
I'm reading through this again and I am still not comfortable being the one to make a decision on this modification. I have a few questions.

If the trim control is attached to the trim tab on the elevator, which is modeled in this simulation.
Does the elevator move at all with trim adjustment? Why?
Does the Yoke move at all with trim adjustment? Why?

Depending on those questions and answers, why wouldn't we rather add some FDM to capture the trim tab adjustments?

**Comment by wlbragg** [comment]
I think I just answered my own questions after thinking this through a little more.
You state
>it has direct mechanical link (cables) between flight controls and control surfaces

So the added pressure, or lack thereof, changes the displacement of the elevator which  in turn displaces the yoke?

If this is so, how much range can the elevator trim affect the elevator surface?

**Comment by wlbragg** [comment]
One more step here, if all of the above is true, shouldn't it be logical that a trim tab adjustment move both the elevator and the yoke slightly? How is this PR different from that?

**Comment by Murmur79** [comment]
Hello,

to be clear, this "fix" is (can be) completely unrelated to the other change I proposed ("Modeling stick-force-per-'g'").

As it is now, using the elevator trim in the default C172, physically moves the elevator surfaces (as apparent in the external view), but does **NOT** move the yoke (as apparent in the cockpit view).

Now, this is physically wrong: yoke and elevator are physically connected in the C172, so there is a biunivocal direct relationship between the two.

This happens because currently the visuals for the flight surfaces (elevator, rudder, etc.) are using the correct variables, whereas the visuals for the yoke and pedals are using the wrong variables.

So, I'm just proposing to use the correct variables also for the yoke and pedal positions.

This would assure that yoke and pedal positions are always in accordance with the _actual_ positions of the flight controls surfaces, _regardless_ of how one decides to simulate the actual dynamics and interaction of flight controls with g-load, airspeed, etc. (as I proposed in the other thread).

In other words, this fix only corrects a visual discrepancy (yoke and pedal positions), and does not in any way affect flight model or anything else.

**Comment by wlbragg** [comment]
@Murmur79 I was just getting ready to merge, after looking at the code, I don't see the necessity for an opt-in/out option. But I noticed you also did the roll and yaw, why is that? I'm not very FDM fluent, but we don't even have a aileron trim, why do we have a roll trim sum? We do have the rudder trim option so I get why there is a yaw trim sum, but not why you moved the connection to the pedals other than for the same reason you did the yoke, because technically they should also be tied and animated with the yaw trim sum and not the rudder controls directly?

**Comment by hbeni** [comment]
> If the trim control is attached to the trim tab on the elevator, which is modeled in this simulation. Does the elevator move at all with trim adjustment? Why? Does the Yoke move at all with trim adjustment? Why?

Yes, in flight the air will move over the trim tab, which produces a force pulling the elevator up (or down).
This also will move the yoke, because it is mechanically linked to the elevator.

Just at the ground, trimming has no effect on the elevator position, because no airflow happens (unless you have wind).
So, to summarize: I think we can live with that and merge.

Regarding the aileron-trim: This might be to have the code consistent. However, we should not reflect the aileron-trim onto the animation: like statet, the c172 has no aileron trim. The trimming in roll is factory-set by adjustint mouting angles if the wings. This is already simulated by a fixed aileron-trim setting in our fdm, but again: ailerons animation should not reflect this.

**Comment by Murmur79** [comment]
>  > Regarding the aileron-trim: This might be to have the code consistent. However, we should not reflect the aileron-trim onto the animation: like statet, the c172 has no aileron trim. The trimming in roll is factory-set by adjustint mouting angles if the wings. This is already simulated by a fixed aileron-trim setting in our fdm, but again: ailerons animation should not reflect this.

The fact is that the "trim sum" variables are the fdm variables that determine the actual position of the flight surfaces. This is true even if there is no trim for a given axis.

So I think it would be consistent to use the "trim sum" variable also for the roll/aileron axis.

**Comment by hbeni** [comment]
The yoke should be perfectly centered with the currently fixed trim.
So that part of the fix is wrong…
Reasoning: the currently fdm-side hard ailerin-trim setting simulates the asymetrical mount of the wings. This should not be reflected in the yokes rotation!

**Comment by Murmur79** [comment]
> The yoke should be perfectly centered with the currently fixed trim. So that part of the fix is wrong… Reasoning: the currently fdm-side hard ailerin-trim setting simulates the asymetrical mount of the wings. This should not be reflected in the yokes rotation!

That is a good point! Indeed, a fixed trim implemented by an asymmetrical mount of the wings ( but not a fixed trim implemented with a fixed trim surface!) is the only case in which this is true.

So you are right, in the case of the C172, we should not change the variable for the  yaw axis

`<channel name="Yaw">`

but only for the two remaining axes

```
<summer name="Pitch Trim Sum">
<summer name="Roll Trim Sum">
```

**Comment by hbeni** [comment]
@wlbragg was merged, can be closed
